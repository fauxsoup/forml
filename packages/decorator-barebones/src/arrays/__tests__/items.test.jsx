import Items from '../items';
import React from 'react';
import { render } from '@testing-library/react';
import Context from '@forml/context';

describe('renders', function () {
    let localizer;
    beforeEach(function () {
        localizer = { getLocalizedString: jest.fn((id) => id) };
    });
    test('its children', function () {
        const { container } = render(
            <Context.Provider value={{ localizer }}>
                <Items>
                    <div id="test">test</div>
                </Items>
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
        expect(container.querySelector('#test')).not.toBeNull();
        expect(localizer.getLocalizedString).toHaveBeenCalled();
    });
});
