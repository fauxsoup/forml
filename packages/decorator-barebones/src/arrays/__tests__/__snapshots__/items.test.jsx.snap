// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders its children 1`] = `
<div>
  <div
    class="array"
  >
    <button
      class="add"
    >
      Add
    </button>
    <ul>
      <div
        id="test"
      >
        test
      </div>
    </ul>
  </div>
</div>
`;
